nix/p2pcollab-bloomf.nix

```diff
@@ -0,0 +1,72 @@
+{ version
+, homepage
+, downloadPage
+, changelog
+, maintainers
+, platforms
+, runFullTestSuite ? false
+}:
+{ lib
+, nix-filter
+, ocamlPackages
+, ...
+}:
+
+let
+  inherit (nix-filter) inDirectory;
+
+  pname = "bloomf";
+  duneTestCommand = if runFullTestSuite then "build @runtest-rand" else "runtest";
+in
+ocamlPackages.buildDunePackage {
+  inherit pname version;
+
+  src = nix-filter {
+    root = ./..;
+    include = [
+      "${pname}.opam"
+      "dune-project"
+      (inDirectory "src")
+      (inDirectory "test")
+    ];
+    name = "p2pcollab-${pname}";
+  };
+
+  minimumOCamlVersion = "4.03";
+  useDune2 = true;
+
+  dontPatch = true;
+  dontConfigure = true;
+
+  buildInputs = with ocamlPackages; [
+    bitv.out
+  ];
+
+  doCheck = true;
+  checkInputs = with ocamlPackages; [
+    alcotest.out
+  ];
+  checkPhase = ''
+    runHook preCheck
+    dune ${duneTestCommand} -p ${pname} ''${enableParallelBuilding:+-j $NIX_BUILD_CORES}
+    runHook postCheck
+  '';
+
+  meta = {
+    description = "Efficient Bloom filters for OCaml";
+    longDescription =
```

Comment:
I think that a multiline Nix string would work here, like:
```
longDescription = ''
  some text
  across multiple
  lines
'';
```

Reply:
I initially adapted this method when I was trying to stay below a certain line length. But then I realized being strict about this kinda also harm the readability by creating verbose expressions. I relaxed the line length rule on most places but I guess `longDescription` must have stayed like that.

The idea was to avoid line breaks that don't appear in the source text. When I do something like
```
longDescription = ''
  Bloom filters are memory and time efficient data structures allowing
  probabilistic membership queries in a set.\n A query negative result ensures
  that the element is not present in the set, while a positive result might be
  a false positive, i.e. the element might not be present and the BF membership
  query can return true anyway.\n Internal parameters of the BF allow to
  control its false positive rate depending on the expected number of elements
  in it.
``;
```
the resulting string has unwanted line breaks.

I guess we could do 
```
longDescription = ''
  Bloom filters are memory and time efficient data structures allowing probabilistic membership queries in a set.
  A query negative result ensures that the element is not present in the set, while a positive result might be a false positive, i.e. the element might not be present and the BF membership query can return true anyway.
  Internal parameters of the BF allow to control its false positive rate depending on the expected number of elements in it.``;
```
If the length of the line is not a problem.